code/game/objects/items/blueprints.dm

```diff
@@ -238,19 +238,17 @@
 		return
 
 	//stuff tied to the area to rename
-	var/list/to_rename = list(
+	var/static/list/to_rename = typecacheof(list(
 		/obj/machinery/airalarm,
 		/obj/machinery/atmospherics/components/unary/vent_scrubber,
 		/obj/machinery/atmospherics/components/unary/vent_pump,
 		/obj/machinery/door,
 		/obj/machinery/firealarm,
 		/obj/machinery/light_switch,
 		/obj/machinery/power/apc,
-	)
-
-	for(var/turf/area_turf as anything in area.get_contained_turfs())
-		for(var/obj/machine in area_turf)
-			if(!is_type_in_list(machine, to_rename))
-				continue
-			machine.name = replacetext(machine.name, oldtitle, title)
+	))
+	for (var/list/zlevel_turfs as anything in area.get_zlevel_turf_lists())
+		for (var/turf/area_turf as anything in zlevel_turfs)
```

Comment:
the UX on this kinda sucks I ain't gonna lie to you, do you think there's any merit in returning a flattened list or does that make for loop copying too big an issue

Reply:
there is a flattened list variant `get_turfs_from_all_zlevels()` no real comment on which one to use but `get_zlevel_turf_lists()` is the easiest to use if you want the split lists.

it really boils down to if there is a chance it could be used on a large area like space or outdoors on icebox.